pkg/sdk/resource_monitors.go

```diff
@@ -75,29 +221,131 @@ func (v *resourceMonitors) Create(ctx context.Context, id AccountObjectIdentifie
+}
+
+type NotifiedUser struct {
+	Name string `ddl:"keyword,double_quotes"`
+}
+
+type Frequency string
+
+func FrequencyFromString(s string) (*Frequency, error) {
+	s = strings.ToUpper(s)
+	if monthly := string(FrequencyMonthly); monthly == s {
+		return (*Frequency)(&monthly), nil
+	}
+	if daily := string(FrequencyDaily); daily == s {
+		return (*Frequency)(&daily), nil
+	}
+	if weekly := string(FrequencyWeekly); weekly == s {
+		return (*Frequency)(&weekly), nil
+	}
+	if yearly := string(FrequencyYearly); yearly == s {
+		return (*Frequency)(&yearly), nil
+	}
+	if never := string(FrequencyNever); never == s {
+		return (*Frequency)(&never), nil
+	}
+	return nil, fmt.Errorf("Invalid frequency type: %s", s)
+}
+
+const (
+	FrequencyMonthly Frequency = "MONTHLY"
+	FrequencyDaily   Frequency = "DAILY"
+	FrequencyWeekly  Frequency = "WEEKLY"
+	FrequencyYearly  Frequency = "YEARLY"
+	FrequencyNever   Frequency = "NEVER"
+)
+
 // AlterResourceMonitorOptions contains options for altering a resource monitor.
-type AlterResourceMonitorOptions struct{}
+type AlterResourceMonitorOptions struct {
+	alter           bool                    `ddl:"static" sql:"ALTER"`            //lint:ignore U1000 This is used in the ddl tag
+	resourceMonitor bool                    `ddl:"static" sql:"RESOURCE MONITOR"` //lint:ignore U1000 This is used in the ddl tag
+	IfExists        *bool                   `ddl:"keyword" sql:"IF EXISTS"`
+	name            AccountObjectIdentifier `ddl:"identifier"`
+	Set             *ResourceMonitorSet     `ddl:"keyword" sql:"SET"`
+	NotifyUsers     *NotifyUsers            `ddl:"parameter,equals" sql:"NOTIFY_USERS"`
+	Triggers        *[]TriggerDefinition    `ddl:"keyword,no_comma" sql:"TRIGGERS"`
+}
+
+func (opts *AlterResourceMonitorOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return ErrInvalidObjectIdentifier
+	}
+	if opts.Set == nil {
+		return nil
+	}
+	if (opts.Set.Frequency != nil && opts.Set.StartTimeStamp == nil) || (opts.Set.Frequency == nil && opts.Set.StartTimeStamp != nil) {
+		return errors.New("must specify frequency and start time together")
+	}
 
-func (v *resourceMonitors) Alter(ctx context.Context, id AccountObjectIdentifier, opts *AlterResourceMonitorOptions) error {
 	return nil
 }
 
+func (v *resourceMonitors) Alter(ctx context.Context, id AccountObjectIdentifier, opts *AlterResourceMonitorOptions) error {
+	if opts == nil {
+		opts = &AlterResourceMonitorOptions{}
+	}
+	opts.name = id
+
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type ResourceMonitorSet struct {
+	// at least one
+	CreditQuota    *int       `ddl:"parameter,equals" sql:"CREDIT_QUOTA"`
+	Frequency      *Frequency `ddl:"parameter,equals" sql:"FREQUENCY"`
+	StartTimeStamp *string    `ddl:"parameter,equals,single_quotes" sql:"START_TIMESTAMP"`
```

Comment:
should this be *time.Time?

Reply:
I did initially use time.Time here, but I think the issue of timestamps is quite tricky.

Reply:
I did initially use time.Time here, but I think the issue of timestamps is quite tricky.

Reply:
okay fair point. String is a good choice if "IMMEDIATELY" is one of the options.